authentication/middleware.go

```diff
@@ -0,0 +1,56 @@
+package authentication
+
+import (
+	"errors"
+	"net/http"
+	"strings"
+
+	jwt "github.com/dgrijalva/jwt-go"
+)
+
+const headerKey = "Authorization"
+
+var (
+	errInvalidHeader = errors.New("invalid authorization header")
+)
+
+// Middleware -
+type Middleware interface {
+	ServeHTTP(w http.ResponseWriter, r *http.Request, next http.HandlerFunc)
+}
+
+type middleware struct {
+	secret []byte
+}
```

Comment:
I think that for most of our middlewares we could just implement for [`negroni.UseFunc`](https://godoc.org/github.com/urfave/negroni#Negroni.UseFunc) instead of [`negroni.Use`](https://godoc.org/github.com/urfave/negroni#Negroni.Use). That would save quite a lot of code, improving readability IMO. No `struct` or `interface` would be necessary.